src/core/base/basePickBy.js

```diff
@@ -0,0 +1,28 @@
+import baseGet from './baseGet'
+import baseSet from './baseSet'
+import getSize from '../util/getSize'
+import getKey from '../util/getKey'
+import setKey from '../util/setKey'
+import castPath from '../util/castPath'
+import withMutations from '../with/withMutations'
+import { hintConvert, get } from '../'
+
+const setPaths = withMutations((result, data, paths, predicate) => {
+  const length = getSize(paths)
+  let index = -1
+
+  while (++index < length) {
+    const path = get(paths, index)
+    const value = baseGet(data, path, getKey)
+    if (predicate(value, path)) {
```

Comment:
Yeah, this part looks a little fishy to me too. Basically, I want to loop thru the paths and check if the value exists in that paths.  Looking at the Lodash implementation:

```
while (++index < length) {
  var path = paths[index],
  value = baseGet(object, path);
  if (predicate(value, path) {
   ...
  }
}
```

What's a better way to implement this so that it will work with immutable data? :)